core/src/main/java/org/apache/accumulo/core/classloader/DefaultContextClassLoaderFactory.java

```diff
@@ -63,7 +64,12 @@ public DefaultContextClassLoaderFactory(final AccumuloConfiguration accConf) {
 
   private static void startCleanupThread(final AccumuloConfiguration conf,
       final Supplier<Map<String,String>> contextConfigSupplier) {
-    ScheduledFuture<?> future = ThreadPools.createGeneralScheduledExecutorService(conf)
+    ScheduledFuture<?> future = ThreadPools.getClientThreadPools(new UncaughtExceptionHandler() {
+      @Override
+      public void uncaughtException(Thread arg0, Throwable arg1) {
+        LOG.error("context classloader cleanup thread has failed.", arg1);
+      }
+    }).createGeneralScheduledExecutorService(conf)
```

Comment:
Formatting is probably bad, but should be possible to use a lambda here.

```suggestion
    ScheduledFuture<?> future = ThreadPools.getClientThreadPools((t, e) ->
        LOG.error("context classloader cleanup thread has failed.", e)).createGeneralScheduledExecutorService(conf)
```

Reply:
Addressed in ec0e7cf